runestone/fitb/js/fitb.js

```diff
@@ -222,23 +222,24 @@ export default class FITB extends RunestoneBase {
       data.sid = sid;
       feedback = false;
     }
+    // Per `logBookEvent <logBookEvent>`, the result is undefined if there's no server. Otherwise, the server provides the endpoint-specific results in ``data.details``; see `make_json_response`.
     data = await this.logBookEvent(data);
-    data = data.detail;
+    let detail = data && data.detail;
```

Comment:
It's typical JS shorthand. If data is defined and is an object, then its value is [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy). Since `true && expression == expression`, this is a nice way to access `data.detail` if and only if `data` is defined.

Reply:
I figured that out, its just not an idiom I've seen in the wild before.

Reply:
JS: always another obscure expression to overcome awkward syntax... :(